packages/wdio-sauce-service/src/service.ts

```diff
@@ -102,23 +103,26 @@ export default class SauceService implements Services.ServiceInstance {
         }
 
         if (this._browser && !this._isJobNameSet) {
+            let jobName = this._suiteTitle
             if (this._options.setJobName) {
-                let jobName = this._options.setJobName(
+                jobName = this._options.setJobName(
                     this._config,
                     this._capabilities,
                     this._suiteTitle!
                 )
-                this._browser.execute(`sauce:job-name=${jobName}`)
+                await this.setAnnotation(`sauce:job-name=${jobName}`)
```

Comment:
Let's switch the flag here otherwise we call `setAnnotation` twice.
```suggestion
                this._isJobNameSet = true
                await this.setAnnotation(`sauce:job-name=${jobName}`)
                
```

Reply:
Thanks, pushed a fixe